nemo_aligner/models/nlp/gpt/megatron_gpt_dpo_model.py

```diff
@@ -361,8 +363,23 @@ def get_ref_policy_logprobs(self, batch):
         pos_ids = torch.cat((batch["position_ids"], batch["position_ids"]), dim=0)
         labels = torch.cat((batch["chosen_labels"], batch["rejected_labels"]), dim=0)
         global_batch = [tokens, masks, pos_ids, labels]
-        with cpu_weight_swap(self, self.ref_policy_state_dict, megatron_amp_O2=self.megatron_amp_O2):
+
+        if self.use_peft and self.ref_policy_state_dict is None:
+            # disable adapters -> reference model
+            for _, module in self.named_modules():
```

Comment:
can we make this a context manager? just like `cpu_weight_swap` but `with disable_lora_adaptors` in utils.utils since other methods might need this too

Reply:
make sense https://github.com/NVIDIA/NeMo-Aligner/pull/142/commits/4237f5eb311fc097ce5b032e3cba5df7201c70cd